deployment/terraform/keycloak.go

```diff
@@ -0,0 +1,279 @@
+		if err != nil {
+			return fmt.Errorf("failed to upload keycloak realm file: %w", err)
+		}
+		extraArguments = append(extraArguments, "--import-realm")
+	} else if t.config.ExternalAuthProviderSettings.KeycloakDBDumpURI == "" {
+		mlog.Info("No realm file or database dump provided, using loadtest's default keycloak realm configuration")
+
+		keycloakRealmFile, err := assets.AssetString("mattermost-realm.json")
+		if err != nil {
+			return fmt.Errorf("failed to read default keycloak realm file: %w", err)
+		}
+
+		_, err = sshc.Upload(strings.NewReader(keycloakRealmFile), "/opt/keycloak/keycloak-"+t.config.ExternalAuthProviderSettings.KeycloakVersion+"/data/import/mattermost-realm.json", true)
+		if err != nil {
+			return fmt.Errorf("failed to upload default keycloak realm file: %w", err)
+		}
+		extraArguments = append(extraArguments, "--import-realm")
+	}
+
+	// Values for the keycloak.env file
+	keycloakEnvFileContents := []string{
+		// Enable health endpoints
+		"KC_HEALTH_ENABLED=true",
+		// Setup admin user
+		"KEYCLOAK_ADMIN=" + t.config.ExternalAuthProviderSettings.KeycloakAdminUser,
+		"KEYCLOAK_ADMIN_PASSWORD=" + t.config.ExternalAuthProviderSettings.KeycloakAdminPassword,
+		// Ensure Java JVM has enough memory for large imports
+		"JAVA_OPTS=-Xms1024m -Xmx2048m",
+		// Logging
+		"KC_LOG_FILE=/opt/keycloak/keycloak-" + t.config.ExternalAuthProviderSettings.KeycloakVersion + "/data/log/keycloak.log",
+		"KC_LOG_FILE_OUTPUT=json",
+		// Database
+		"KC_DB_POOL_MIN_SIZE=20",
+		"KC_DB_POOL_INITIAL_SIZE=20",
+		"KC_DB_POOL_MAX_SIZE=200",
+		"KC_DB=postgres",
+		"KC_DB_URL=jdbc:psql://localhost:5432/keycloak",
+		"KC_DB_PASSWORD=mmpass",
+		"KC_DB_USERNAME=keycloak",
+		"KC_DATABASE=keycloak",
+	}
+
+	// Upload keycloak.env file
+	_, err = sshc.Upload(strings.NewReader(strings.Join(keycloakEnvFileContents, "\n")), "/etc/systemd/system/keycloak.env", true)
+	if err != nil {
+		return fmt.Errorf("failed to upload keycloak env file: %w", err)
+	}
+
+	// Parse keycloak service file template
+	tmpl, err := template.New("keycloakServiceFile").Parse(string(keycloakServiceFileContents))
+	if err != nil {
+		return fmt.Errorf("failed to parse keycloak service file template: %w", err)
+	}
+	var buf bytes.Buffer
+	err = tmpl.Execute(&buf, struct {
+		KeycloakVersion string
+		Command         string
+	}{
+		KeycloakVersion: t.config.ExternalAuthProviderSettings.KeycloakVersion,
+		Command:         command + " " + strings.Join(extraArguments, " "),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to execute keycloak service file template: %w", err)
+	}
+
+	// Install systemd service
+	_, err = sshc.Upload(&buf, "/etc/systemd/system/keycloak.service", true)
+	if err != nil {
+		return fmt.Errorf("failed to upload keycloak service file: %w", err)
+	}
+
+	// Enable and start service
+	_, err = sshc.RunCommand("sudo systemctl enable keycloak")
```

Comment:
Don't you need to do a `sudo systemctl daemon-reload` before enabling? Usually we do that for other services in the test.

Reply:
You are right, though I have created environments from scratch that started the service correctly, but a `daemon-reload` should be required. Mhmmm. Maybe is only required for changes but not for new units? I need to check this.